### PR TITLE
Support interactive output for fuzzing.

### LINKED ISSUE #2395: Support interactive mode for clusterfuzz.fuzz
Where output is streamed to console rather than dumped all at once.

### PR DESCRIPTION
**Pull request by oliverchang**
This will be used for libClusterFuzz.

Fuzzing output will be dumped to stdout in real time.

Fixes #2395.

### CODE REVIEW

File: src/clusterfuzz/__init__.py [hunk 1]
```diff
@@ -12,3 +12,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """libClusterfuzz package."""
+
+import os
+
+if not os.getenv('ROOT_DIR'):
+  # libClusterFuzz
+  os.environ['CONFIG_DIR_OVERRIDE'] = os.path.join(
+      os.path.dirname(os.path.abspath(__file__)), 'lib-config')
```
Comment (inferno-chromium) [inline src/clusterfuzz/__init__.py:21]: nit: move os.path.dirname(os.path.abspath(__file__) to local and use in both calls.

File: src/clusterfuzz/__init__.py [hunk 2]
```diff
@@ -12,3 +12,18 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """libClusterfuzz package."""
+
+import os
+
+if not os.getenv('ROOT_DIR'):
+  # libClusterFuzz
```
Comment (inferno-chromium) [inline src/clusterfuzz/__init__.py:19]: can you explain this comment a little more.

File: src/clusterfuzz/_internal/system/environment.py [hunk 3]
```diff
@@ -1076,4 +1076,4 @@ def is_android_kernel(plt=None):
 
 def is_lib():
   """Whether or not we're in libClusterFuzz."""
-  return get_value('ROOT_DIR') is None
+  return get_value('LIB_CF')
```
Comment (inferno-chromium) [inline src/clusterfuzz/_internal/system/environment.py:1079]: bool() ?

File: src/clusterfuzz/fuzz/__main__.py [hunk 4]
```diff
@@ -31,7 +30,7 @@ def main():
       '-e',
       '--engine',
       help='Fuzzing engine.',
-      choices=['libFuzzer', 'honggfuzz'],
+      choices=['libFuzzer', 'afl', 'honggfuzz'],
```
Comment (inferno-chromium) [inline src/clusterfuzz/fuzz/__main__.py:33]: can we add this somewhere in constants and use in both places OR best to derive it from some fuzzing engines integrated in cf.

File: src/clusterfuzz/_internal/system/new_process.py [hunk 5]
```diff
@@ -270,8 +299,18 @@ def run(self,
...
+
+    stdout_file = None
     if stdout == subprocess.PIPE and max_stdout_len:
-      stdout = tempfile.TemporaryFile()
+      stdout_file = tempfile.TemporaryFile()
+
+    interactive = environment.get_value('CF_INTERACTIVE')
+    if interactive:
+      popen_args['bufsize'] = 0
+      if stdout != subprocess.PIPE:
```
Comment (inferno-chromium) [inline src/clusterfuzz/_internal/system/new_process.py:310]: Can you add a comment why ?

**Review by inferno-chromium — APPROVED** [review]

**Review by jonathanmetzman — APPROVED** [review]
LGTM

File: src/clusterfuzz/__init__.py (hunk 1 above)
Comment (oliverchang) [reply src/clusterfuzz/__init__.py:21]: done!

File: src/clusterfuzz/__init__.py (hunk 2 above)
Comment (oliverchang) [reply src/clusterfuzz/__init__.py:19]: done!

File: src/clusterfuzz/_internal/system/environment.py (hunk 3 above)
Comment (oliverchang) [reply src/clusterfuzz/_internal/system/environment.py:1079]: get_value already does this.

File: src/clusterfuzz/fuzz/__main__.py (hunk 4 above)
Comment (oliverchang) [reply src/clusterfuzz/fuzz/__main__.py:33]: Will do this in another PR to keep this one clean.

**Pull request merged**